dummycloud/src/MqttClient.js

```diff
@@ -19,10 +19,22 @@ class MqttClient {
             this.handleData(data);
         })
     }
-    
+
     initialize() {
-        this.client = mqtt.connect(process.env.MQTT_BROKER_URL);
-        
+        const options = {
+            clientId: "deye_dummycloud" + Math.random().toString(16).slice(2, 10),
+        };
+
+        if (process.env.MQTT_USERNAME && process.env.MQTT_PASSWORD) {
+            options.username = process.env.MQTT_USERNAME;
+            options.password = process.env.MQTT_PASSWORD;
+        } else if (process.env.MQTT_USERNAME || process.env.MQTT_PASSWORD) {
+            Logger.error("Both MQTT_USERNAME and MQTT_PASSWORD need to be set.");
```

Comment:
Funfact: Actually this isn't required by the MQTT Spec. That's one of the interesting quirks of the protocol

See https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html

```
[MQTT-3.1.2-18]
If the User Name Flag is set to 0, a user name MUST NOT be present in the payload.

[MQTT-3.1.2-19]
If the User Name Flag is set to 1, a user name MUST be present in the payload.

[MQTT-3.1.2-20]
If the Password Flag is set to 0, a password MUST NOT be present in the payload.

[MQTT-3.1.2-21]
If the Password Flag is set to 1, a password MUST be present in the payload.

[MQTT-3.1.2-22]
If the User Name Flag is set to 0, the Password Flag MUST be set to 0.

```

While it does say that the Password Flag being 1 requires the User Name Flag to be 1, it's not requiring it the other way round